dlrs/main/classes/RollupSummariesSelector.cls

```diff
@@ -363,37 +427,167 @@ public class RollupSummariesSelector {
+          this.getLookupRollupSummariesWithParameters(
+            newQueryFactory()
+              .addOrdering(
+                LookupRollupSummary2__mdt.Label,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .toSOQL(),
+            new List<Object>()
+          )
+        );
+      } else {
+        List<RollupSummary> summaries = RollupSummary.toList(
+          this.getLookupRollupSummaries()
+        );
+
+        return sortSummaries(summaries, 'Label', null);
+      }
+    }
+
+    protected virtual List<SObject> getLookupRollupSummariesWithParameters(
+      String queryString,
+      List<Object> tmpVars
+    ) {
+      //Executing dynamic SOQL requires any variable binding to be in the same
+      //scope. We only ever use 2 variables in this class, so just hard coding
+      //two temporary variables for the dynamic SOQL to run
+      Object tmpVar1 = tmpVars.size() > 0 ? tmpVars[0] : null;
+      Object tmpVar2 = tmpVars.size() > 1 ? tmpVars[1] : null;
+
+      return Database.query(queryString);
+    }
+
+    protected virtual List<SObject> getLookupRollupSummaries() {
+      return LookupRollupSummary2__mdt.getAll().values();
+    }
+
+    protected virtual SObject getLookupRollupSummaryByDeveloperNameOrId(
+      String developerNameOrId
+    ) {
+      return LookupRollupSummary2__mdt.getInstance(developerNameOrId);
+    }
+
+    private List<RollupSummary> sortSummaries(
+      List<RollupSummary> summaries,
+      String sortField,
+      String sortFieldSecondary
+    ) {
+      List<RollupSummaryComparable> rollupSummaryComparables = new List<RollupSummaryComparable>();
+      for (RollupSummary summary : summaries) {
+        rollupSummaryComparables.add(
+          new RollupSummaryComparable(summary, sortField, sortFieldSecondary)
+        );
+      }
+
+      rollupSummaryComparables.sort();
+
+      List<RollupSummary> sortedSummaries = new List<RollupSummary>();
+
+      for (RollupSummaryComparable comparable : rollupSummaryComparables) {
+        sortedSummaries.add(comparable.summary);
+      }
+
+      return sortedSummaries;
+    }
+  }
+
+  private class RollupSummaryComparable implements Comparable {
+    private RollupSummary summary;
+    private String fieldToSortOn;
+    private String sortOrder;
+    private String fieldToSortOnSecondary;
+
+    private RollupSummaryComparable(
+      RollupSummary summary,
+      String fieldToSortOn,
+      String fieldToSortOnSecondary
+    ) {
+      this.summary = summary;
+      this.fieldToSortOn = fieldToSortOn;
+      this.fieldToSortOnSecondary = fieldToSortOnSecondary;
+    }
+
+    public Integer compareTo(Object compareTo) {
+      RollupSummaryComparable compareToSummary = (RollupSummaryComparable) compareTo;
+
+      // The return value of 0 indicates that both elements are equal.
+      Integer returnValue = 0;
+      if (
+        String.valueOf(this.summary.record.get(this.fieldToSortOn)) >
```

Comment:
extract to local vars maybe? would improve readability 
Not sure we can implement a switch statement here but those if/elifs make it hard to grasp the intention